src/components/navitem/Navitem.tsx

```diff
@@ -0,0 +1,52 @@
+import { FC } from "react";
+import { useMatch } from "react-router-dom";
+import {
+  NavIcon,
+  NavItemContainer,
+  NavLinkStyled,
+  NavText,
+} from "./NavItemStyledComponents";
+
+export interface NavItemProps {
+  permission: string;
+  text: string;
+  Icon?: React.ComponentType<any>;
+  className?: string;
+  isExpanded: boolean;
+}
+
+export interface IClientStyling {
+  thumbnail: string;
+  logo: string;
+}
+
+const Navitem: FC<NavItemProps> = ({ permission, text, Icon, isExpanded }) => {
```

Comment:
can we update this prop to href or something like navigateTo ?
```suggestion
const Navitem: FC<NavItemProps> = ({ href, text, Icon, isExpanded }) => {
```